clang/lib/Driver/ToolChains/ClassicFlang.cpp

```diff
@@ -91,7 +97,7 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
   // Add input file name to the compilation line
   const char *InputFile = Input.getBaseInput();
   if (strcmp(InputFile, "-") == 0)
-    UpperCmdArgs.push_back("/dev/stdin");
+    UpperCmdArgs.push_back(NULL_FILE);
```

Comment:
You've just changed `/dev/stdin` to `/dev/null`....
The `HIP.cpp` I have mentioned in one of my commends was just an example how problems with Windows are handled in other parts of Clang, I did not mention that this is the final solution.

For me, this code should look like:
```
#if defined(_WIN32) || defined(_WIN64)
  UpperCmdArgs.push_back(NULL_FILE);
#else
  UpperCmdArgs.push_back("/dev/stdin");
#endif
```

Reply:
it makes more sense than replacing stdin with null (for non-Windows case).

If you want to avoid doubling, try this instead:
```
#if defined(_WIN32) || defined(_WIN64)
  UpperCmdArgs.push_back("nul");
#else
  UpperCmdArgs.push_back("/dev/stdin");
#endif
```
...then you can remove previous NULL_FILE definition, won't be needed anymore.

Reply:
@pawosm-arm done